src/Mod/Draft/draftguitools/gui_hyperlink.py

```diff
@@ -0,0 +1,116 @@
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+"""Provides GUI tools to open hyperlinks to internal/external documents."""
+
+## @package gui_hyperlink
+# \ingroup draftguitools
+# \brief Provides GUI tools to open hyperlinks to internal/external documents
+
+## \addtogroup draftguitools
+# @{
+
+import FreeCAD
+import os, sys, subprocess
+import re
+import draftutils.utils as utils
+from draftutils.messages import _msg
+
+if FreeCAD.GuiUp:
+    import FreeCADGui
+    import Draft_rc
+def QT_TRANSLATE_NOOP(ctx,txt):
+    return txt
+translate = FreeCAD.Qt.translate
+
+from PySide import QtWidgets
+
+__title__ = "FreeCAD Draft Workbench GUI Tools - Hyperlinks tools"
+__author__ = ("")
+__url__ = "https://www.freecad.org"
+
+
+class Draft_Hyperlink:
+    """The Draft_Hyperlink FreeCAD command definition."""
+
+    def GetResources(self):
+        d = {'Pixmap': '',
+             'Accel': "",
+             'MenuText': QT_TRANSLATE_NOOP("Draft_Hyperlink", "Open hyperlinks"),
+             'ToolTip': QT_TRANSLATE_NOOP("Draft_Hyperlink", "Open linked documents")}
+        return d
+
+    def Activated(self):
+        self.find_hyperlinks()
+
+        if len(self.hyperlinks_list) > 1:
+            m = QtWidgets.QMessageBox()
+            m.setWindowTitle(translate("draft", "Opening multiple hyperlinks"))
+            m.setText(
+                translate(
+                    "draft",
+                    "Multiple hyperlinks found."
+                )
+            )
+            m.setInformativeText(
+                translate(
+                    "draft",
+                    "This may lead to the opening of various windows"
+                )
+            )
+            m.setStandardButtons(m.Ok | m.Cancel)
+            ret = m.exec_()
+
+        if len(self.hyperlinks_list) == 1 or ret == m.Ok:
+            for hyperlink in self.hyperlinks_list:
+                self.open_hyperlink(hyperlink)
+
+    def find_hyperlinks(self):
+        self.hyperlinks_list = []
+
+        for o in FreeCADGui.Selection.getCompleteSelection():
+            if hasattr(o.Object, "Text"):
+
+                for text in o.Object.Text:
+                    hyperlinks = re.findall(r"((%\w+%|/\w+|\w:\\|\w{3,4}://)[\w\\/: ]+\.[\w?$#=\\\.]+)", text)
```

Comment:
You have mentioned certain issues with the regular expression (https://github.com/FreeCAD/FreeCAD/issues/7062#issuecomment-1742073962). And I am not sure what is meant by 'internal/external document' in the feature request. But here are some more examples that the regular expression does not handle:
```
text = "File on server: \\Server2\Share\Test\Foo.txt"
text = "Alternative Windows path: D:/path/file.ext"
text = "Remote file: https://forum.freecad.org/viewtopic.php?f=23&t=69528"
```
Sadly I am not a regex specialist (you know much more than I do). Maybe one of the other devs has a suggestion, or else you could ask in the [Developers corner](https://forum.freecad.org/viewforum.php?f=10)